### DISCUSSION TITLE
How to access nested zones?

### DISCUSSION THREAD

**Discussion by willnationsdev** [OrchardCMS/OrchardCore#8575] [Q&A] [answered]
Our theme has...

- Layout.cshtml
    - `RenderBodyAsync()`
- Content.cshtml
    - `RenderSectionAsync("Footer")`

How would one go about accessing the nested zone? The `<zone>` tag helper always jumps to the ThemeLayout zones.

I want to define a shape template in the Admin Templates GUI that will appear in the "Footer" region, but I'm not sure how to target it since it isn't a content type, part, field, or layout zone.

**Comment by Skrypt** [comment, marked as answer]
You can create an **admin template** named "Layout" but then it will override the entire admin UI and you will need to create a Liquid template that is at least equivalent to what we have with our Layout.cshtml file. Now, I think this could work but I'm not sure we could inject the current dark mode theme then.

Needs to be tested.

Then, of course you could add your custom zones.

For now, the quickest way to do this is to create an admin theme and change the Layout.cshtml file to suit your needs.

**Reply by willnationsdev** [reply]
@Skrypt 
I've gone ahead and refactored all of my Content.cshtml to be part of the Layout.cshtml (and thereby removed the need for a Content.cshtml). That has solved our problem.

However, related, if I look at the Content.cshtml in the OrchardCore.Contents module, it renders things like a Header, a Meta, and a Footer, etc. I initially assumed those were zones, but now that I think about it, they are part of the ContentItem's Model. How do you even define IHtmlContent for those properties?

**Reply by willnationsdev** [reply]
After investigating the code a bit more, I have a feeling that these are areas of code that DisplayDrivers, with their `.Location("SomeZone")`, can use to target the creation of shapes in those areas, rather than them being delivered from properties. For example, I noticed that the Content.SummaryAdmin.cshtml template includes a reference to rendering `Model.ActionsMenu`, but nothing appears to have that as a property. So, instead, it's rather than display drivers can append their own content to those sections of the page.

I wasted a lot of time trying to find any reference to an "ActionsMenu" string in a DisplayType parameter somewhere before I finally realized what was going on.

So, the answer to the original question, rendering to nested zones, is that you have to be rendering a shape that is a child of whatever is rendering the zone, and then you need to use a placement on it to force its positioning in that zone. That's what it seems like anyway.

**Reply by Skrypt** [reply]
> it renders things like a Header, a Meta, and a Footer, etc. I initially assumed those were zones

Yes, these are Shape Locations which are defined in the ContentsDriver.cs file of the OrchardCore.Contents module for example.

> How do you even define IHtmlContent for those properties?

These are rendered by using the `DisplayHelper.ShapeExecuteAsync(shape)` method from within the RazorPage.cs extension that is defined in the OrchardCore.DisplayManagement assembly. That's basically what is the OC DisplayManagement.

> So, the answer to the original question, rendering to nested zones, is that you have to be rendering a shape that is a child of whatever is rendering the zone, and then you need to use a placement on it to force its positioning in that zone. That's what it seems like anyway.

Yes, adding shapes to a Location combined with a Placement will add it to the `List<IDisplayResult>();` of a driver. Though, it's not really "nesting" unless you are adding a sub Shape within a Shape for example.
